main.tf

```diff
@@ -100,7 +114,7 @@ resource "ibm_is_vpc" "vpc" {
 resource "ibm_is_vpc_dns_resolution_binding" "vpc_dns_resolution_binding_id" {
   count = (var.enable_hub == false && var.enable_hub_vpc_id) ? 1 : 0
 
-  name   = "${var.prefix}-dns-binding"
+  name   = var.prefix != null ? "${var.prefix}-dns-binding" : var.dns_binding_name
```

Comment:
Regardless of the value of `var.prefix`, if the user passes a value for `var.dns_binding_name` the code should use that. Otherwise use "${var.prefix}-${var.name}-dns-binding" unless prefix is null, in which case use "${var.name}-dns-binding"

Reply:
Now changed it to the following. It will always check first if `dns_binding_name` is there or not, if no then check with prefix or name. 

```
name   = var.dns_binding_name != null ? var.dns_binding_name : (var.prefix != null ? "${var.prefix}-dns-binding" : "${var.name}-dns-binding")
```